modules/netns-isolation.nix

```diff
@@ -241,66 +265,40 @@ in {
         "127.0.0.1"
       ];
       bitcoind-host = netns.bitcoind.address;
-      cli = pkgs.writeScriptBin "lncli"
-      # Switch user because lnd makes datadir contents readable by user only
-      ''
-        netns-exec nb-lnd sudo -u lnd ${config.services.lnd.package}/bin/lncli --tlscertpath ${config.nix-bitcoin.secretsDir}/lnd-cert \
-          --macaroonpath '${config.services.lnd.dataDir}/chain/bitcoin/mainnet/admin.macaroon' "$@"
-      '';
+      cliExec = mkCliExec "lnd";
     };
 
-    # liquidd: Custom netns configs
-    services.liquidd = mkIf config.services.liquidd.enable {
```

Comment:
Hey, I'm afraid these `mkIf`s are necessary since many of the options only exist if the modules are enabled. Otherwise you get an `error: attribute 'lnd' missing` for example.

Reply:
It works, because, thanks to nix' lazyness, the option values are never evaluated for disabled modules.

Reply:
Hmm, I'm getting `error: attribute 'lnd' missing` when enabling only clightning. How can that be?

Reply:
I can't reproduce this.
Here's a script that successfully evals a secure-node config with only clightning enabled. Run it from within nix-shell.
```bash
read -d '' tmpstr <<'EOF'
(import <nixpkgs/nixos> {
configuration = { config, lib, ... }: with lib; {
    imports = [
      <nix-bitcoin/modules/presets/secure-node.nix>
    ];

    nix-bitcoin.netns-isolation.enable = true;

    services.clightning.enable = true;
    # services.spark-wallet.enable = true;
    # services.lightning-charge.enable = true;
    # services.nanopos.enable = true;
    # services.lnd.enable = true;
    # services.lnd.listenPort = 9736;
    # services.lightning-loop.enable = true;
    # systemd.services.lightning-loop.wantedBy = mkForce [];
    # services.electrs.enable = true;
   # services.backups.enable = true;
  };
}).vm.outPath
EOF
nix eval "($tmpstr)"
```

Reply:
You're probably referencing `services.lnd.listen` somewhere in your config, even when `lnd` is disabled.

Reply:
Hmm, your script does run fine, however I'm getting

```
while evaluating 'mkCliExec' at /nix-bitcoin/modules/netns-isolation.nix:46:15, called from /nix-bitcoin/modules/n:
attribute 'lnd' missing, at /nix-bitcoin/modules/netns-isolation.nix:46:43
```

when deploying with `nixops deploy`

Reply:
Fixed. The bug was in webindex.